File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/application/ports/SenhaHistoricoRepository.java
package br.ufpr.sept.so2.modules.iam.application.ports;

import java.util.List;
import java.util.UUID;

public interface SenhaHistoricoRepository {

    void append(UUID usuarioId, String senhaHash);

    List<String> findLastHashes(UUID usuarioId, int limite);
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/application/ports/SenhaHistoricoRepository.kt
package br.ufpr.sept.so2.modules.iam.application.ports

import java.util.UUID

interface SenhaHistoricoRepository {
    fun append(usuarioId: UUID, senhaHash: String)

    fun findLastHashes(usuarioId: UUID, limite: Int): List<String>
}
